### PR TITLE
Global DAO Alerts for Realms Notification across via Notifi Provider

### PR DESCRIPTION
**Pull request by Nathan-Notifi**
- creates global alerts for every source via Notifi sdk
- updates notification switch to handle preview of data
- creates custom phone input
- allows users to update and delete alerts per dao

# First Time User
ftu

# Updating Alerts
updating alerts

# Updating contacts
Updating contacts

# Updating confirmed vs unconfirmed
Updating contacts

### CODE REVIEW

File: components/NotificationsCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+} from 'react'
+import { useCallback } from 'react'
+
+import useWalletStore from '../../stores/useWalletStore'
+import Button from '../Button'
+import Input from '../inputs/Input'
+import NotifiFullLogo from './NotifiFullLogo'
+import PhoneInput from './PhoneInput'
+
+const firstOrNull = <T,>(
```
Comment (nramadas) [inline components/NotificationsCard/index.tsx:28]: if we move this to `utils/`, we can reuse that version here

**Review by nramadas — CHANGES_REQUESTED** [review]
hey Nathan-Notifi, i'm a new member at Solana workig on realms, thought I'd take a look at the PR. Just a few comments from my end, otherwise things look good!

File: components/NotificationsCard/NotificationCardContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    setTelegramEnabled(telegram !== undefined)
+  }, [getConfiguration, setTelegramEnabled])
+
+  useEffect(() => {
+    updateTelegramSupported().catch((e) => {
+      console.error('Failed to get supported type information: ', e)
+    })
+  }, [updateTelegramSupported])
+
+  const firstOrNull = <T,>(
```
[the remark this answers is not in the document]
Comment (Nathan-Notifi) [reply components/NotificationsCard/NotificationCardContainer.tsx:84]: sounds good. How about adding it to `helpers`?

File: components/NotificationsCard/PhoneInput.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    [dialCode, handlePhone]
+  )
+
+  const splitPhoneNumbers = (phoneNumber: string) => {
+    if (!isValidPhoneNumber(phoneNumber)) {
+      return
+    }
+    const { baseNumber, countryCode } = splitPhoneNumber(phoneNumber)
+    if (!countryCode || !baseNumber) {
+      throw new Error('Improper phone')
```
[the remark this answers is not in the document]
Comment (Nathan-Notifi) [reply components/NotificationsCard/PhoneInput.tsx:55]: yeah, i updated it to just display on the input

File: components/NotificationsCard/NotifiPreviewCard.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+}) => {
+  const alerts = data?.alerts
+  const sources = data?.sources
+  const [isLoading, setLoading] = useState(false)
+
+  const handleEdit = useCallback(() => {
+    onClick()
+  }, [onClick])
+
+  const Line = () => (
```
[the remark this answers is not in the document]
Comment (Nathan-Notifi) [reply components/NotificationsCard/NotifiPreviewCard.tsx:46]: you bet

**Pull request merged**